tests/unit/plugins/modules/test_omevv_firmware.py

```diff
@@ -0,0 +1,259 @@
+                                       "parameter must be between 60 and 1440."
+CLUSTER_HOST_SERVICETAG_MUTUAL_EXCLUSIVE_MSG = "parameters are mutually " \
+                                               "exclusive: cluster|host|servicetag."
+CLUSTER_HOST_SERVICETSAG_REQUIRED_MSG = "Either 'cluster' or 'host' or 'servicetag' must " \
+                                        "be specified."
+UPDATE_JOB_PRESENT_MSG = "Update job is either running or in a scheduled state for cluster " \
+                         "'{cluster_name}'. Wait for its completion and trigger."
+JOB_NAME_ALREADY_EXISTS_MSG = "Job with name '{job_name}' already exists. Provide different name."
+CLUSTER_HOST_NOT_FOUND_MSG = "No managed hosts found in the cluster."
+HOST_NOT_FOUND_MSG = "Host '{managed_host}' not found under managed hosts."
+CLUSTER_NOT_FOUND_MSG = "Provided cluster name '{cluster_name}' is not valid."
+CHANGES_FOUND_MSG = "Changes found to be applied."
+CHANGES_NOT_FOUND_MSG = "No changes found to be applied."
+TIMEOUT_NEGATIVE_OR_ZERO_MSG = "The value for the 'job_wait_timeout' parameter cannot be " \
+                               "negative or zero."
+UNREACHABLE_MSG = "The URL with the {ip}:{port} cannot be reached."
+SOURCE_NOT_FOUND_MSG = "The Requested resource cannot be found."
+TRIGGER_UPDATE_CHECK_URI = "/Consoles/{vcenter_uuid}/CanTriggerUpdate"
+
+
+class TestFirmwareUpdate(FakeAnsibleModule):
+    module = omevv_firmware
+
+    @pytest.fixture
+    def omevv_firmware_mock(self):
+        omevv_obj = MagicMock()
+        return omevv_obj
+
+    @pytest.fixture
+    def omevv_connection_firmware(self, mocker, omevv_firmware_mock):
+        omevv_conn_mock = mocker.patch(MODULE_PATH + 'RestOMEVV',
+                                       return_value=omevv_firmware_mock)
+        omevv_conn_mock.return_value.__enter__.return_value = omevv_firmware_mock
+        return omevv_conn_mock
+
+    # def test_execute(self, omevv_default_args, omevv_connection_firmware_repository_profile):
+    #     pass
+    #     # obj = MagicMock()
+    #     # omevv_obj = self.module.FirmwareRepositoryProfile(
+    #     #     omevv_connection_firmware_repository_profile, obj)
+    #     # omevv_obj.execute()
+
+    def test_get_payload_details(self, mocker, omevv_connection_firmware, omevv_default_args):
+        # Scenario 1: payload details with description
+        host_id = 123456
+        obj = MagicMock()
+        omevv_obj = self.module.FirmwareUpdate(
+            omevv_connection_firmware, obj)
+        mocker.patch(MODULE_PATH + 'FirmwareUpdate.add_optional_fields', return_value=None)
+        mocker.patch(MODULE_PATH + 'FirmwareUpdate.set_schedule', return_value=None)
+        mocker.patch(MODULE_PATH + 'FirmwareUpdate.set_job_details', return_value=None)
+        mocker.patch(MODULE_PATH + 'FirmwareUpdate.add_targets', return_value=None)
+        result = omevv_obj.get_payload_details(host_id)
+        assert "firmware" in result
```

Comment:
We should be doing assertion to check the exact return of the method `get_payload_details` based on the input/mocking. Please update

Reply:
Addressed the comment